tests/e2e-pw/specs/shopper/shopper-checkout-cart-coupon.spec.ts

```diff
@@ -0,0 +1,73 @@
+/**
+ * External dependencies
+ */
+import test, { expect } from '@playwright/test';
+
+/**
+ * Internal dependencies
+ */
+import { config } from '../../config/default';
+import {
+	goToCart,
+	goToCheckout,
+	goToShop,
+} from '../../utils/shopper-navigation';
+import { useShopper } from '../../utils/helpers';
+import {
+	addToCartFromShopPage,
+	emptyCart,
+	fillBillingAddress,
+	fillCardDetails,
+	placeOrder,
+	setupCheckout,
+} from '../../utils/shopper';
+
+const productName = config.products.simple.name;
+
+test.describe(
+	'Checkout with free coupon & after modifying cart on Checkout page',
+	() => {
+		// All tests will use the shopper only.
+		useShopper();
+
+		test.beforeEach( async ( { page } ) => {
+			await goToShop( page );
+			await addToCartFromShopPage( page, productName );
+			await goToCart( page );
+			await page.getByPlaceholder( 'Coupon code' ).fill( 'free' );
+			await page.getByRole( 'button', { name: 'Apply coupon' } ).click();
+		} );
+
+		test( 'Checkout with a free coupon', async ( { page } ) => {
+			await goToCheckout( page );
+			await fillBillingAddress( page, config.addresses.customer.billing );
+			await placeOrder( page );
+			await page.waitForURL( /\/order-received\//, {
+				waitUntil: 'load',
+			} );
+			await expect(
+				page.getByRole( 'heading', {
+					name: 'Order received',
+				} )
+			).toBeVisible();
+			await emptyCart( page );
```

Comment:
Shouldn't the cart be empty after placing an order? 🤔

Reply:
Sorry, those comments somehow were left in the "Pending" state, I thought you missed them and continued to merge. Then I realized I needed to submit the review to make them visible to you (which I don't know why).